packages/cdk/src/config.ts

```diff
@@ -0,0 +1,214 @@
+  assertValidExternalSecret(externalSecret);
+  const { system, key, type } = externalSecret;
+  let rawValue: ExternalSecretPrimitive;
+  switch (system) {
+    case 'aws_ssm_parameter_store': {
+      const [paramPath, paramKey] = key.split(':');
+      if (!(paramPath && paramKey)) {
+        throw new OperationOutcomeError(
+          validationError(
+            'Keys for AWS Param Store secrets must be a path followed by a ":" and a key. (/path/to/param:my_key)'
+          )
+        );
+      }
+      rawValue = await fetchParameterStoreSecret(paramPath, paramKey);
+      break;
+    }
+    default:
+      throw new OperationOutcomeError(
+        validationError(`Unknown system '${system}' for ExternalSecret. Unable to fetch the secret for key '${key}'.`)
+      );
+  }
+  return normalizeFetchedValue(key, rawValue, type);
+}
+
+async function normalizeInfraConfigArray(
+  currentVal: any[]
+): Promise<ExternalSecretPrimitive[] | Record<string, any>[]> {
+  // ------ case 3a: primitives or `ExternalSecret`
+  const firstEle = currentVal[0];
+  let newArray: ExternalSecretPrimitive[] | Record<string, any>[];
+  if ((typeof firstEle !== 'object' && firstEle !== null) || isExternalSecretLike(firstEle)) {
+    newArray = new Array(currentVal.length) as ExternalSecretPrimitive[];
+    for (let i = 0; i < currentVal.length; i++) {
+      const currIdxVal = currentVal[i] as unknown as ExternalSecretPrimitive | ExternalSecret;
+      if (typeof currIdxVal !== 'object') {
+        newArray[i] = currIdxVal;
+        continue;
+      }
+      const fetchedVal = await fetchExternalSecret(currIdxVal);
+      newArray[i] = fetchedVal;
+    }
+  }
+  // ------ case 3b: other objects (recurse)
+  else {
+    newArray = new Array(currentVal.length) as Record<string, any>[];
+    for (let i = 0; i < currentVal.length; i++) {
+      newArray[i] = await normalizeObjectInInfraConfig(currentVal[i]);
+    }
+  }
+  return newArray;
+}
+
+async function normalizeValueForKey(obj: Record<string, any>, key: string): Promise<any> {
+  const currentVal = obj[key];
+  // cases:
+  // --- case 1: primitive
+  if (typeof currentVal !== 'object') {
+    obj[key] = currentVal;
+  }
+  // --- case 2: object conforming to `ExternalSecret` schema
+  else if (isExternalSecretLike(currentVal)) {
+    obj[key] = await fetchExternalSecret(currentVal);
+  }
+  // --- case 3: an array of:
+  else if (Array.isArray(currentVal) && currentVal.length) {
+    obj[key] = await normalizeInfraConfigArray(currentVal);
+  }
+  // --- case 4: other object (recurse)
+  else if (typeof currentVal === 'object') {
+    obj[key] = await normalizeObjectInInfraConfig(currentVal);
+  }
+}
+
+export async function normalizeObjectInInfraConfig(obj: Record<string, any>): Promise<Record<string, any>> {
+  const normalizedObj = { ...obj };
+  // walk config object
+  for (const key in normalizedObj) {
+    if (Object.hasOwn(normalizedObj, key)) {
```

Comment:
You can skip the `Object.hasOwn()` call if you iterate over `Object.keys` such as

```ts
for (const key of Object.keys(normalizedObj)) {
```